front/app/components/admin/PostManager/components/PostPreview/Idea/Components/OfflineVoteSettings.tsx

```diff
@@ -0,0 +1,204 @@
+import React, { useMemo } from 'react';
+
+import {
+  Box,
+  Image,
+  Input,
+  Text,
+  Tooltip,
+} from '@citizenlab/cl2-component-library';
+import { debounce } from 'lodash-es';
+
+import useAppConfiguration from 'api/app_configuration/useAppConfiguration';
+import useIdeaById from 'api/ideas/useIdeaById';
+import useUpdateIdea from 'api/ideas/useUpdateIdea';
+import useAuthUser from 'api/me/useAuthUser';
+import { VotingMethod } from 'api/phases/types';
+import usePhase from 'api/phases/usePhase';
+import useUpdatePhase from 'api/phases/useUpdatePhase';
+import useUserById from 'api/users/useUserById';
+
+import { trackEventByName } from 'utils/analytics';
+import { FormattedMessage } from 'utils/cl-intl';
+import { isNilOrError } from 'utils/helperUtils';
+
+import messages from '../messages';
+import tracks from '../tracks';
+
+import tooltipPreview from './assets/preview_for_tooltip.png';
+import ListItemWithBullet from './ListItemWithBullet';
+
+interface Props {
+  ideaId: string;
+  phaseId: string;
+  votingMethod: VotingMethod;
+}
+
+const OfflineVoteSettings = ({ ideaId, votingMethod, phaseId }: Props) => {
+  const { data: authUser } = useAuthUser();
+
+  const { data: appConfig } = useAppConfiguration();
+  const { data: phase } = usePhase(phaseId);
+  const tenantId = appConfig?.data.id;
+  const { data: idea } = useIdeaById(ideaId);
+  const { data: userLastModifiedVotes } = useUserById(
+    idea?.data.relationships.manual_votes_last_updated_by?.data?.id
+  );
+  const { data: userLastModifiedVoters } = useUserById(
+    phase?.data.relationships.manual_voters_last_updated_by?.data?.id
+  );
+  const { mutate: updateIdea } = useUpdateIdea();
```

Comment:
Not necessarily caused by this PR, but when I was testing updating the manual votes I saw that a bunch of `ideas_count` calls were being invalidated and refetched each time the number changes:
![Screenshot 2024-11-04 at 20 26 44](https://github.com/user-attachments/assets/621604e2-2d26-43a3-8e6b-3da583992da9)

Not sure why those would need to be refetched when a single idea changes... maybe we could remove this cache invalidation in `useUpdateIdea`. But not really a critical issue, just nice to have

Reply:
I _think_ we might need these invalidations depending on how the Idea is updated :thinking: E.g. If the idea is moved between phases (Idea `phase_ids` changes), we need to invalidate the counts maybe?

Though if we're just doing simple updates to an idea, I agree that it shouldn't trigger all these cache invalidations. I guess we would want some sort of conditional cache invalidation given a certain payload, though I'm not sure we're doing this anywhere currently..

@luucvanderzee Do you have any thoughts here? Maybe something to bring up at the next code discussion meeting? :thinking:

Reply:
> I guess we would want some sort of conditional cache invalidation given a certain payload, though I'm not sure we're doing this anywhere currently..

This is actually quite easy! For example, for `useUpdateIdea` you can update the update object like this:

```
type IUpdateIdeaObject = {
  id: string;
  requestBody: IIdeaUpdate;
  doNotRefetchCounts?: boolean // New attribute
};
```

Then you can use the second argument of `onSuccess` to access this variable:

```
onSuccess: (idea, variables) => {
...
if (!variables.doNotRefetchCounts) {
  queryClient.invalidateQueries({ queryKey: ideaFilterCountsKeys.all() });
  queryClient.invalidateQueries({ queryKey: ideasCountKeys.items() });
}
```

and then when you call `updateIdea` you just add `doNotRefetchCounts: true` to the object. Probably the naming and stuff could be improved here but that's the general logic. Pretty sure we do this in more places

Reply:
Ahh smart - ok I'll take a look into doing this :) Thanks!

Reply:
You could also derive this from the `requestBody` by checking which parameters have changed btw- e.g. if only the `manual_votes` changed do not refetch. But this might be more complicated, not sure